src/views/index-dtf/issuance/async-swaps/index.tsx

```diff
@@ -0,0 +1,183 @@
+import { Button } from '@/components/ui/button'
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
+import useERC20Balance from '@/hooks/useERC20Balance'
+import { cn } from '@/lib/utils'
+import { isSafeMultisigAtom } from '@/state/atoms'
+import { indexDTFAtom } from '@/state/dtf/atoms'
+import { useAtom, useAtomValue, useSetAtom } from 'jotai'
+import { ArrowLeft, Settings } from 'lucide-react'
+import { useEffect } from 'react'
+import useTrackIndexDTFPage from '../../hooks/useTrackIndexDTFPage'
+import RefreshQuote from '../../overview/components/zap-mint/refresh-quote'
+import ZapSettings from '../../overview/components/zap-mint/zap-settings'
+import AsyncMint from './async-mint'
+import AsyncRedeem from './async-redeem'
+import {
+  asyncSwapFetchingAtom,
+  asyncSwapInputAtom,
+  asyncSwapOngoingTxAtom,
+  asyncSwapRefetchAtom,
+  currentAsyncSwapTabAtom,
+  defaultSelectedTokenAtom,
+  indexDTFBalanceAtom,
+  mintTxHashAtom,
+  selectedTokenAtom,
+  showAsyncSwapSettingsAtom,
+} from './atom'
+import Collaterals from './collaterals'
+import GnosisSafeRequired from './gnosis-safe-required'
+import OrderStatusUpdater from './order-status-updater'
+import Success from './success'
+import Updater from '../manual/updater'
+
+const AsyncSwaps = () => {
+  useTrackIndexDTFPage('mint-async-swap')
+  const isSafeMultisig = useAtomValue(isSafeMultisigAtom)
+  const [currentTab, setCurrentTab] = useAtom(currentAsyncSwapTabAtom)
+  const [showSettings, setShowSettings] = useAtom(showAsyncSwapSettingsAtom)
+  const defaultToken = useAtomValue(defaultSelectedTokenAtom)
+  const setSelectedToken = useSetAtom(selectedTokenAtom)
+  const indexDTF = useAtomValue(indexDTFAtom)
+  const asyncSwapRefetch = useAtomValue(asyncSwapRefetchAtom)
+  const asyncSwapFetching = useAtomValue(asyncSwapFetchingAtom)
+  const asyncSwapOngoingTx = useAtomValue(asyncSwapOngoingTxAtom)
+  const input = useAtomValue(asyncSwapInputAtom)
+  const setIndexDTFBalance = useSetAtom(indexDTFBalanceAtom)
+  const mintTxHash = useAtomValue(mintTxHashAtom)
+  const invalidInput = isNaN(Number(input)) || Number(input) === 0
+
+  const { data: balance } = useERC20Balance(indexDTF?.id)
+
+  useEffect(() => {
+    setIndexDTFBalance(balance || 0n)
+  }, [balance, setIndexDTFBalance])
+
+  const reset = () => {
+    setShowSettings(false)
+    setSelectedToken(defaultToken)
+  }
+
+  const changeTab = (tab: string) => {
+    setCurrentTab(tab as 'mint' | 'redeem')
+  }
+
+  useEffect(() => {
+    return () => {
+      reset()
+    }
+  }, [])
+
+  if (!indexDTF) return null
+
+  if (isSafeMultisig) {
+    return (
+      <div className="container flex flex-col items-center sm:justify-start md:justify-center gap-2 lg:border-2 lg:border-secondary lg:bg-secondary/30 lg:h-[calc(100vh-100px)] dark:bg-card rounded-4xl w-full">
```

Comment:
the same container div is used 3 times, create a Container component with the variations instead